controller/src/main/java/org/jboss/as/controller/audit/InMemoryAuditLogHander.java

```diff
@@ -61,33 +61,31 @@
     private int maxHistory;
     private final AuditLogItemFormatter myFormatter = new InMemoryFormatter();
 
+
     public InMemoryAuditLogHander(String name, int maxHistory) {
         super(name, IN_MEMORY_FORMATTER_NAME, maxHistory);
-        items = new ArrayList<>(maxHistory);
+        this.items = new ArrayList<>(maxHistory);
         this.maxHistory = maxHistory;
         setFormatter(myFormatter);
     }
 
-    public List<ModelNode> getItems() {
+    @Override
+    public List<ModelNode> listLastEntries() {
         return Collections.unmodifiableList(items);
     }
 
-    public void setMaxHistory(int maxHistory) {
-        synchronized (items) {
-            this.maxHistory = maxHistory;
-            while (maxHistory < items.size()) {
-                items.remove(0);
-            }
+    @Override
+    public void setMaxFailureCount(int maxHistory) {
+        super.setMaxFailureCount(maxHistory);
+        this.maxHistory = maxHistory;
+        while (maxHistory < items.size()) {
+            items.remove(0);
         }
     }
```

Comment:
Looking at ManagedAuditLogger, it already has handler-specific modification methods, e.g. updateSyslogHandlerAppName, updateSyslogHandlerFacility, updateSyslogHandlerReconnectTimeout. That's a bit yuck, but since they are already there I don't see why another one can't be added for setting the max history.

ManagedAuditLogger isn't usable outside the kernel so we can be a bit casual about its API.